### ISSUE TITLE
BCFIPS not supprted with pulsar client libraray

### ISSUE THREAD

**Issue by kemburi** [apache/pulsar#6517]
**Describe the bug**
Need bcfips support for pulsar clinet.

**To Reproduce**
Just deploy on a jboss server which has bcfips support.
Pulsar client will fail with NoClassDefFoundError:

**Expected behavior**
there should be an option to provde a security provider. like bcfips.

**Additional context**
java.lang.NoClassDefFoundError: Could not initialize class org.apache.pulsar.common.util.SecurityUtility
This class has a static block which load the bc provider.
Here is the code block. Here we want to load org.bouncycastle.jce.provider.BouncyCastleFIPSProvider.

static {
        // Fixes loading PKCS8Key file: https://stackoverflow.com/a/18912362
        java.security.Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
    }

**Comment by yjshen** [comment]
I suppose this should be a feature request?

**Comment by kemburi** [comment]
could be, we have an issue where our jboss server already has bcfips jars included, hence this client itself failing, if there is an option to add bcfipis as some parameter, i am fine to accept this as feature.
Is there any configuration to load a particular secuirty provider?

**Comment by kemburi** [comment]
static {
// Fixes loading PKCS8Key file: https://stackoverflow.com/a/18912362
java.security.Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
}
Above hardcoding mandates that we should always use  org.bouncycastle.jce.provider.BouncyCastleProvider(), which i think is a bug, it could read from a system property which would become configurable.
This will impact anyone who is already on some other providers.

**Comment by jiazhai** [comment]
will take a look

**Comment by jiazhai** [comment]
@kemburi #6588 merged for fix this issue.

**Comment by kemburi** [comment]
@jiazhai 
can someone do a maven release of this so that we can consume into our projects?

**Comment by jiazhai** [comment]
@kemburi 2.5.1 release will be cut out this week. mean while, you can try StreamNative weekly build:
```
    <repository>
      <id>bintray-streamnative-maven</id>
      <name>bintray</name>
      <url>https://dl.bintray.com/streamnative/maven</url>
    </repository>
```

this release contains this feature. https://github.com/streamnative/pulsar/releases/tag/v2.6.0-98344a711

**Comment by jiazhai** [comment]
would like to close this issue, since fixed in 2.5.1
